src/opentelemetry/_opamp/client.py

```diff
@@ -0,0 +1,109 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from logging import getLogger
+from typing import Generator
+
+from uuid_utils import uuid7
+from opentelemetry.util.types import AnyValue
+
+import opentelemetry._opamp.messages as messages
+from opentelemetry._opamp.transport.requests import RequestsTransport
+from opentelemetry._opamp.version import __version__
+from opentelemetry._opamp.proto import opamp_pb2
+
+
+_logger = getLogger(__name__)
+
+_DEFAULT_OPAMP_TIMEOUT_MS = 1_000
+
+_OTLP_HTTP_HEADERS = {
+    "Content-Type": "application/x-protobuf",
+    "User-Agent": "OTel-OpAMP-Python/" + __version__,
+}
+
+_HANDLED_CAPABILITIES = (
+    opamp_pb2.AgentCapabilities.AgentCapabilities_ReportsStatus
+    | opamp_pb2.AgentCapabilities.AgentCapabilities_ReportsHeartbeat
+    | opamp_pb2.AgentCapabilities.AgentCapabilities_AcceptsRemoteConfig
+)
+
+
+class OpAMPClient:
+    def __init__(
+        self,
+        *,
+        endpoint: str,
+        headers: dict[str, str] | None = None,
+        timeout_millis: int = _DEFAULT_OPAMP_TIMEOUT_MS,
+        agent_identifying_attributes: dict[str, AnyValue],
+        agent_non_identifying_attributes: dict[str, AnyValue] | None = None,
+    ):
+        self._timeout_millis = timeout_millis
+        self._transport = RequestsTransport()
+
+        self._endpoint = endpoint
+        headers = headers or {}
+        self._headers = {**_OTLP_HTTP_HEADERS, **headers}
+
+        self._agent_description = messages._build_agent_description(
+            identifying_attributes=agent_identifying_attributes,
+            non_identifying_attributes=agent_non_identifying_attributes,
+        )
+        self._sequence_num: int = 0
+        self._instance_uid: bytes = uuid7().bytes
+
+    def _build_connection_message(self) -> bytes:
+        message = messages._build_presentation_message(
+            instance_uid=self._instance_uid,
+            agent_description=self._agent_description,
+            sequence_num=self._sequence_num,
+            capabilities=_HANDLED_CAPABILITIES,
+        )
+        data = messages._encode_message(message)
+        return data
+
+    def _build_agent_disconnect_message(self) -> bytes:
+        message = messages._build_agent_disconnect_message(
+            instance_uid=self._instance_uid,
+            sequence_num=self._sequence_num,
+        )
+        data = messages._encode_message(message)
+        return data
+
+    def _build_heartbeat_message(self) -> bytes:
+        message = messages._build_heartbeat_message(instance_uid=self._instance_uid, sequence_num=self._sequence_num)
```

Comment:
Should this include the `capabilities` field? 
https://github.com/open-telemetry/opamp-spec/blob/main/specification.md#agenttoservercapabilities says:

> This field MUST be always set.

Though IIRC our current OpAMP server doesn't care. I don't think it is looking at AgentToServer.capabilities.